src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-config-modal/rgw-config-modal.component.ts

```diff
@@ -68,43 +73,50 @@ export class RgwConfigModalComponent implements OnInit {
   }
 
   checkKmsProviders() {
-    this.kmsProviders = rgwBucketEncryptionModel.kmsProviders;
+    const selectedEncryptionType = this.configForm.get('encryptionType').value;
+    //  for kms
+    if (selectedEncryptionType === ENCRYPTION_TYPE.SSE_KMS) {
+      this.kmsProviders = [KMS_PROVIDER.VAULT, KMS_PROVIDER.KMIP];
+    } else if (selectedEncryptionType === ENCRYPTION_TYPE.SSE_S3) {
+      this.kmsProviders = [KMS_PROVIDER.VAULT];
+    }
     if (
       this.allEncryptionConfigValues &&
-      this.allEncryptionConfigValues.hasOwnProperty('SSE_KMS') &&
+      this.allEncryptionConfigValues.hasOwnProperty(ENCRYPTION_TYPE.SSE_KMS) &&
       !this.editing
     ) {
-      const sseKmsBackends = this.allEncryptionConfigValues['SSE_KMS'].map(
+      const kmsBackends = Object.values(this.allEncryptionConfigValues[ENCRYPTION_TYPE.SSE_KMS]).map(
         (config: any) => config.backend
       );
-      if (this.configForm.get('encryptionType').value === rgwBucketEncryptionModel.SSE_KMS) {
+      if (this.configForm.get('encryptionType').value === ENCRYPTION_TYPE.SSE_KMS) {
         this.kmsProviders = this.kmsProviders.filter(
-          (provider) => !sseKmsBackends.includes(provider)
+          (provider) => !kmsBackends.includes(provider)
         );
       }
     }
     if (
       this.allEncryptionConfigValues &&
-      this.allEncryptionConfigValues.hasOwnProperty('SSE_S3') &&
+      this.allEncryptionConfigValues.hasOwnProperty('s3') &&
       !this.editing
     ) {
-      const sseS3Backends = this.allEncryptionConfigValues['SSE_S3'].map(
+      const s3Backends = Object.values(this.allEncryptionConfigValues[ENCRYPTION_TYPE.SSE_S3]).map(
         (config: any) => config.backend
       );
-      if (this.configForm.get('encryptionType').value === rgwBucketEncryptionModel.SSE_S3) {
+      if (this.configForm.get('encryptionType').value === ENCRYPTION_TYPE.SSE_S3) {
         this.kmsProviders = this.kmsProviders.filter(
-          (provider) => !sseS3Backends.includes(provider)
+          (provider) => !s3Backends.includes(provider)
         );
       }
     }
-    if (this.kmsProviders.length > 0 && !this.kmsProviders.includes('vault')) {
-      this.configForm.get('kms_provider').setValue('');
+    if (this.kmsProviders.length > 0) {
+      this.configForm.get('kms_provider').setValue(this.kmsProviders[0]);
+      this.setKmsProvider();
```

Comment:
you are setting a value for the `kms_provider` control here. Then you are trying to get the same thing inside the `setKmsProvider()` below. Why not simplify them and reduce the logic by just passing the value to that function.
```suggestion
      const kmsProvider = this.kmsProviders[0])
      this.configForm.get('kms_provider').setValue(kmsProvider);
      this.setKmsProvider(kmsProvider);
```

Reply:
changed the function functionality